### PR TITLE
scio 2.13

### LINKED ISSUE #1553: Scio compatibility with Scala 2.13
Scala 2.13 should be released in the next few weeks and could potentially break source compatibility (because of the changes in the Collection API). 

Assuming Scio works properly with Scala 2.12 (https://github.com/spotify/scio/issues/1275), we should make sure it is also compatible with 2.13 and cross-publish binaries for Scala 2.11, 2.12 and 2.13.

- [x] [cats](https://mvnrepository.com/artifact/org.typelevel/cats-core_2.13/2.0.0)
- [x] algebird
- [x] chill
- [x] [jackson-module-scala](https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-scala_2.13/2.9.9)
- [x] [protobuf-generic](https://mvnrepository.com/artifact/me.lyh/protobuf-generic_2.13/0.2.5)
- [x] [case-app](https://mvnrepository.com/artifact/com.github.alexarchambault/case-app_2.13/2.0.0-M9)
- [x] [magnolia](https://mvnrepository.com/artifact/com.propensive/magnolia_2.13/0.11.0) #2192 #2241
- [x] [scalatest](https://mvnrepository.com/artifact/org.scalatest/scalatest_2.13/3.0.8)
- [x] [scalacheck](https://mvnrepository.com/artifact/org.scalacheck/scalacheck_2.13/1.14.0)
- [x] [scalacheck-shapeless](https://mvnrepository.com/artifact/com.github.alexarchambault/scalacheck-shapeless_1.14_2.13/1.2.3)
- [x] [shapeless-datatype](https://mvnrepository.com/artifact/me.lyh/shapeless-datatype-core_2.13/0.2.0)
- [x] [shapeless](https://mvnrepository.com/artifact/com.chuusai/shapeless_2.13/2.3.3)
- [x] annoy4s https://github.com/annoy4s/annoy4s/pull/15
- [x] [breeze](https://mvnrepository.com/artifact/org.scalanlp/breeze_2.13/1.0)
- [x] [circe](https://mvnrepository.com/artifact/io.circe/circe-core_2.13/0.12.1)
- [x] [parquet-avro-extra](https://mvnrepository.com/artifact/me.lyh/parquet-avro-extra_2.13/0.2.3)
- [x] featran
- [x] kantan.csv https://github.com/nrinaudo/kantan.csv/pull/187
- [ ] spark

### LINKED ISSUE #2599: Drop 2.11 support
Due to problems with magnolia and java 11 (#2308) and various other issues. We should start thinking about dropping 2.11 support, regardless of whether we can start supporting 2.13 (#1553)?

### PR DESCRIPTION
**Pull request by regadas**
- [x] scio-core
- [x] scio-avro
- [x] scio-bigtable
- [x] scio-bigquery
- [x] scio-cassandra
- [x] scio-elasticsearch
- [ ] scio-examples
- [x] scio-extra
- [x] scio-jdbc
- [x] scio-macros
- [x] scio-parquet
- [x] scio-repl
- [x] scio-schemas
- [x] scio-smb
- [x] scio-spanner
- [x] scio-sql
- [ ] scio-tensorflow
- [x] scio-test

Fixes #1553 
Fixes #2599

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -147,8 +154,27 @@ lazy val scalafmtSettings = Seq(
 val commonSettings = Sonatype.sonatypeSettings ++ assemblySettings ++ Seq(
   organization := "com.spotify",
   scalaVersion := "2.12.10",
-  crossScalaVersions := Seq("2.11.12", scalaVersion.value),
+  crossScalaVersions := Seq("2.12.10"),
```
Comment (jto) [inline build.sbt]: No 2.13 ?

**Comment by jto** [comment]
Really cool :) 

Is splitting `scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/package.scala ` into multiple files required to migrate to `2.13` or can we do it in another PR ?

**Comment by regadas** [comment]
Hi @jto, yup that's really required for the migration.

**Comment by regadas** [comment]
`scio-tensorflow` and `scio-examples` are the only ones left due to a circular dependency with `featran`. To solve this we need to release the currently supported modules.

**Comment by regadas** [comment]
Ok `CsvIO` introduced some things that i need to refactor

**Pull request merged**